PixivHelper.py

```diff
@@ -59,94 +70,61 @@ def set_log_level(level):
     get_logger(level).setLevel(level)
 
 
-if os.sep == '/':
-    __badchars__ = re.compile(r'^\.|\.$|^ | $|^$|\?|:|<|>|\||\*|\"')
-else:
-    __badchars__ = re.compile(r'^\.|\.$|^ | $|^$|\?|:|<|>|/|\||\*|\"')
-__badnames__ = re.compile(r'(aux|com[1-9]|con|lpt[1-9]|prn)(\.|$)')
-
-__re_manga_index = re.compile(r'_p(\d+)')
-
-
-def sanitize_filename(s, rootDir=None):
+def sanitize_filename(name, rootDir=None):
     '''Replace reserved character/name with underscore (windows), rootDir is not sanitized.'''
     # get the absolute rootdir
     if rootDir is not None:
         rootDir = os.path.abspath(rootDir)
 
     # Unescape '&amp;', '&lt;', and '&gt;'
-    s = html.unescape(s)
+    name = html.unescape(name)
 
-    # Replace badchars and badnames with _
-    name = __badchars__.sub('_', s)
-    if __badnames__.match(name.lower()):
+    # Replace badchars and reserved Windows file names with _
+    name = __badchars__.sub("_", name)
+    if Path(name).is_reserved():
```

Comment:
this won't work if the name contains path separator, you still need to split based on the separator.
```
>>> Path("/aux/new").is_reserved() 
False
>>> os.makedirs("/aux/new")        
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Python38-32\lib\os.py", line 211, in makedirs
    makedirs(head, exist_ok=exist_ok)
  File "C:\Python38-32\lib\os.py", line 221, in makedirs
    mkdir(name, mode)
NotADirectoryError: [WinError 267] The directory name is invalid: '/aux'
```

Reply:
you can refer to https://github.com/Nandaka/PixivUtil2/blob/f1a1b926d2f0873e0d18f4cf45a897fbc6743211/PixivHelper.py#L92 for path split, maybe add the concatenation with '_' there.